From 71a3581f9c0cda06244eae3205ee693d74e648d2 Mon Sep 17 00:00:00 2001
From: danbudris <budris@amazon.com>
Date: Wed, 5 Jan 2022 16:47:51 -0500
Subject: [PATCH] Replace dependencies to mitigate license issues

Replace promptui mod to mitigate LGPL dependency
Troubleshoot has a dependency on the manifoldco/promptui
module, which in turn has a dependency on juju/ansiterm module, which is
LGPLv3 licensed. There is currently a PR to remove this dependency
(https://github.com/manifoldco/promptui/pull/181) that that has not been
merged, but the PR author has updated their own fork to remove this
dependency, so we are using that module instead of the one from manifoldco.

replace xi2/xv to mitigate non-standard license
The xi2/xv package, a dependency of the version of https://github.com/mholt/archiver imported by replicated/troubleshoot, uses a non-standard Public Domain license.
This replacement points to a fork which has a standard CC-0 public domain license.

Signed-off-by: Dan Budris <budris@amazon.com>
---
 go.mod | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/go.mod b/go.mod
index 0f703fd..69066d9 100644
--- a/go.mod
+++ b/go.mod
@@ -193,4 +193,6 @@ require (
 replace (
 	github.com/StackExchange/wmi => github.com/yusufpapurcu/wmi v1.2.2
 	github.com/go-ole/go-ole => github.com/go-ole/go-ole v1.2.6 // needed for arm builds
+	github.com/xi2/xz => github.com/therootcompany/xz v1.0.1
+       github.com/manifoldco/promptui => github.com/nguyer/promptui v0.8.1-0.20210517132806-70ccd4709797
 )
--
2.30.1 (Apple Git-130)
